coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+#Overview
+This code coverage tool has two major features.
+1. In post-submit workflow (when a pull request is merged), it runs code coverage and generates the 
+  following artifacts in a GCS bucket
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+  - code coverage profile, which will be used later in presubmit workflow as a base for comparison
+2. In pre-submit workflow (when a pull requested is created or updated with new commit)
+  - it runs code coverage on target directories and compares the new result with the one stored by 
+  the post-submit workflow and generate coverage difference. It reports coverage changes to the 
+  pull request as a comment by a robot github account. 
+  - it uses go tools to generate line by line coverage and serve the result in html, with a link as 
+  part of the robot comment mentioned above 
+  - it can report a failure status if coverage falls below threshold - this allows repo owner to make the 
+check required and block PR if coverage threshold not met. 
+
+
+##Users
+The presubmit tool is intended for a developer to see the impact on code coverage of his/her commit. It can also be used by repo managers to block any PR from merging if the coverage falls under customized threshold.
+
+The periodical testgrid report is for repo managers and/or test infra team to monitor code coverage stats over time.
+
+##Programming Language Supported
+The code coverage tool only collect code coverage for Go files
+
+
+#Design of Test Coverage Tool
+We pack the test coverage feature in a container, that is triggered to run by a CI/CD system such as prow, in response to Github events such as pulls and merges.The behavior varies in presubmit and postsubmit workflows, which is discussed in latter sections. 
+
+The tool takes input from three sources
+1. It runs test coverage profiling on target repository. 
+2. It reads previously stored post-submit code coverage profile from gcs bucket. The profile
+serves as a base of comparison for presubmit delta coverage.
+3. Variables passed through flags. Those variables include directory to run test coverage, file filters and threshold for desired coverage level.  
```

Comment:
This list is great. I think it would be useful to either describe the configuration options in detail or at a minimum link to where they are declared in the code.